src/gluonts/zebras/_period.py

```diff
@@ -0,0 +1,291 @@
+            return f"{self.multiple}Q"
+
+        np_name, np_multiple = self.np_freq
+        return f"{self.multiple}{self._freq_numpy_to_pandas[np_name]}"
+
+
+@dataclass
+class _BasePeriod:
+    data: np.datetime64
+    multiple: int
+
+    @property
+    def year(self):
+        return self._astype("M8[Y]").astype(np.int16) + 1970
+
+    @property
+    def month(self):
+        return self._astype("M8[M]").astype(np.int8) % 12 + 1
+
+    @property
+    def day(self):
+        return (self.data - self._astype("M8[M]") + 1).astype(np.int8)
+
+    @property
+    def hour(self):
+        return (self.data - self._astype("M8[D]")).astype(np.int8)
+
+    @property
+    def minute(self):
+        return (self.data - self._astype("M8[h]") + 1).astype(np.int8)
+
+    @property
+    def second(self):
+        return (self.data - self._astype("M8[m]")).astype(np.int8)
+
+    @property
+    def dayofweek(self):
+        return (self._astype("M8[D]").astype(int) - 4) % 7
+
+    @property
+    def dayofyear(self):
+        return (self._astype("M8[D]") - self._astype("M8[Y]")).astype(
+            np.int16
+        ) + 1
+
+    @property
+    def week(self):
+        return np.array(
+            [cal.isocalendar().week for cal in self._astype(datetime.datetime)]
+        )
+
+    @property
+    def freq(self):
+        freq = Freq(np.datetime_data(self.data.dtype), self.multiple)
+        return freq.to_pandas()
+
+    @property
+    def __init_passed_kwargs__(self):
+        return asdict(self)
+
+    def to_numpy(self):
+        return self.data
+
+    def __array__(self):
+        return self.data
+
+    def __add__(self, other):
+        if _is_number(other):
+            return self.__class__(
+                self.data + self.multiple * other, multiple=self.multiple
+            )
+
+    def __sub__(self, other):
+        if _is_number(other):
+            return self.__class__(
+                self.data - self.multiple * other, multiple=self.multiple
+            )
+        else:
+            return self.data - other.data
+
+        raise ValueError(other)
+
+
+@functools.total_ordering
+class Period(_BasePeriod):
+    def periods(self, count: int):
+        return _periods(self.data, count, self.multiple)
+
+    def to_pandas(self):
+        import pandas as pd
+
+        return pd.Period(self.data, self.freq)
+
+    def to_timestamp(self):
+        return self.data.astype(object)
+
+    def __repr__(self):
+        return f"Period<{self.data}, {self.freq}>"
+
+    def __lt__(self, other: Period):
+        return self.data < other.data
+
+
+def _periods(start: np.datetime64, count: int, multiple: int) -> Periods:
+    return Periods(np.arange(start, multiple * count, multiple), multiple)
+
+
+class Periods(_BasePeriod):
+    def _astype(self, ty):
+        return self.data.astype(ty)
+
+    def first(self, count: int) -> Periods:
+        return self[:count]
+
+    def last(self, count: int) -> Periods:
+        return self[-count:]
```

Comment:
There should be one, and preferably only one, obvious way to do it!

Reply:
Do you prefer the slicing syntax? Because I don't.